### DISCUSSION TITLE
HP bar with built in playfield border not working as expected

### DISCUSSION THREAD

**Discussion by smoogipoo** [ppy/osu#30880] [Q&A]
### Type

Game behaviour

### Bug description

I understand that this isn't something we "support" per-se. This was reported to me as a **regression** occurring within the last few months. I have **not investigated** this in depth, so this is a tracking issue first to get it out of private DMs.

Transcript:

> I was trying to play on lazer and my skin got completely annihilated in resolution positioning and aspect ratio, more specifically legacy health display, I have my play area border in with my display as 1 file and due to the fact that the skin editor changes the aspect ratio I cannot adjust it in a way that will make it fit as for some reason this file can only just adjusted on the corners

> |  |  |
> | --- | --- |
> | ![image](https://github.com/user-attachments/assets/4a2738cb-a484-4d5d-b7e9-85fe33b3d1e2) | ![image](https://github.com/user-attachments/assets/4229b732-8721-4543-a804-b26bb0da325d) |

> ...
> it was around when pp was able to be gained on lazer
my skins merged and worked
> ...
> its on all my skins
and im pretty sure its due to the aspect ratio of the skin editor
it doesnt preserve the screen ratio while in the editor, if the editing area was the same ratio as the screen i dont think there would be any issue
as i could strech the image out to fit
> ...
> https://drive.google.com/file/d/1-JdjY7UcgPd1ccAaiYIflS63aLf4rGOJ/view?usp=share_link
> https://drive.google.com/file/d/1-ON5S2F84iHHK2bP_MERhSzEZyl-5Bgs/view?usp=share_link
> https://drive.google.com/file/d/1-EhlJCmhVeaPtL_WLOUbdJcNImegkHXD/view?usp=share_link

### Version

2024.1115.3

### Logs

- No logs because this is a forwarded issue.

**Comment by peppy** [comment]
> https://drive.google.com/file/d/1-JdjY7UcgPd1ccAaiYIflS63aLf4rGOJ/view
> https://drive.google.com/file/d/1-EhlJCmhVeaPtL_WLOUbdJcNImegkHXD/view
> https://drive.google.com/file/d/1-ON5S2F84iHHK2bP_MERhSzEZyl-5Bgs/view

These all work fine for me.

The user likely has their UI scale set in settings, which applies to gameplay elements. Should it not? That's up for debate..

@smoogipoo fine to downgrade this to a discussion?

**Reply by fleuph** [reply]
In my opinion, UI scaling should only apply to gameplay elements if you have screen scaling enabled. While HP bar and stuff would definitely be considered "UI" the user-expected behavior for UI scaling probably doesn't include most things that can be edited within the skin editor.
